odoo/fields.py

```diff
@@ -623,6 +623,9 @@ def setup_related(self, model):
             delegate_field = model._fields[self.related.split('.')[0]]
             self._modules = tuple({*self._modules, *delegate_field._modules, *field._modules})
 
+        if self.related and self.store and self.translate:
+            _logger.warning("A related translated field (%s) cannot be stored", self)
```

Comment:
Shouldn't this also unset the `store` or something? Because here it's going to try to create a stored translated related anyway no?

Reply:
I want to leave it to be discussed

I use warning because I assume (maybe I am wrong) many partners' customized codes have translated related stored field. (e.g. related to the product.template.name). But in single language environment, it works as expected. And I don't want to scare them too much and still want the code to be runnable.

Also I guess if a developer sees this warning/error, he will try to make the field computed translated stored, which actually also cannot promise all translations are synchronized. Before we get a perfect solution, I suggest to remain these related translated stored fields for our partners as long as they don't really see any strange thing.

But yes, I agree, technically, it would be better to raise an error.

Reply:
I think the warning is fine and good, it's just that it leaves the system in an incorrect state rather than put it in a less problematic state.

Then again it's not clear what the effect of having a stored translated related is, what happens exactly? Which value gets stored depends on the context lang when resolving the relation?

Reply:
The problem is not about storing such fields.  The problem is that the computation method is not correct: it only (re)computes the field in one single language instead of all languages.  And recomputing such a field in all languages can introduce a performance bottleneck you will not notice unless you test a large database with 10+ languages.  The performance issue is that the compute method will require to fetch the data in every language, as prefetching does not help for this use-case.

Reply:
I agree.
Is this commit message better? Or you want to raise an error here?
```
The _compute_related function can only update one translation but not all.
And we decide not to update all translations to avoid increasing the time complexity.
As a result, the value for related translated fields should always be generated in the runtime, and storing it is illegal.

A related translated stored field may work as expected in a single language
environment. But in mult languages environment, if you have a name field

    name = fields.Char(related="parent_id.name", store=True)

changing parent_id in French, will only change the French translation of the
name field

This PR tries to remove these fields. And a warning is added to prevent
developers creating a related translated stored field in the future.
```